stranding/stranding.py

```diff
@@ -0,0 +1,183 @@
+
+        When `tolerance` is 1.0 and `window` is 0.0 the algorithm will only check for
+        exact sequence matches at the specified coordinates. This is the most performant
+        use case as no alignments are performed.
+
+        Otherwise, the algorithm will load the reference sequences for the 5' and 3'
+        flanks at the specified coordinates extending in each direction extended by
+        `window`. These sequences and their reverse complements are aligned and
+        scored against the query flanks. Alignments scoring above
+        `len(query flank) * match_score * tolerance` are accepted.
+        (a perfect alignment has a score of `len(query flank) * match_score`)
+
+        A return value of 1 indicates that alignments were accepted against the forward
+        reference sequence and the flanks are on the forward strand of the specified
+        reference assembly.
+
+        A return value of -1 indicates that alignments were accepted against the
+        reverse complement of the forward reference sequence and the flanks correspond
+        to the "reverse" or "minus" strand of the specified reference assembly.
+
+        An InconsistentAlignment exception is raised if alignments are accepted on
+        both strands. An Unstrandable exception is raised if no alignments are accepted.
+        """
+        # sanity checks
+        if pos == 0:
+            raise Unstrandable('Position 0 is unmapped')
+        elif chr_name in ('0', 0):
+            raise Unstrandable('Chromosome 0 is unmapped')
+        elif max(len(_5p), len(_3p)) < self.min_flank_length:
+            raise FlanksTooShort('At least one flank must be longer than the specified'
+                                 'minimum flank length of s %d' % self.min_flank_length)
+
+        # chromosome-specific conventions
+        loop = chr_name == 'MT'
+        chr_name = chr_name if chr_name != 'XY' else 'X'
+        max_length = max(len(_5p), len(_3p))
+
+        # reference sequences
+        try:
+            chromosome = Chromosome(chr_name, build, loop=loop)
+            ref_5p = chromosome.sequence(pos - window - max_length, pos + window)
+            ref_3p = chromosome.sequence(pos - window, pos + max_length + window + 1)
+        except ValueError:
+            raise MissingReferenceFlank(
+                'Could not find flanks for %s %d %d' % (chr_name, pos, window))
+
+	# exact comparisons are cheap so try this first
+        if window == 0 and (_3p == ref_3p or _5p == ref_5p):
+            return 1
+
+        ref_5p_RC = str(Seq(ref_5p).reverse_complement())
+        ref_3p_RC = str(Seq(ref_3p).reverse_complement())
+
+        if window == 0 and (_3p == ref_5p_RC or _5p == ref_3p_RC):
+            return -1
+
+        if window == 0 and self.tolerance == 1.0:
+            raise Unstrandable('Strict stranding failed')
+
+        # alignments are expensive so try to do as few as possible
+        fwd_5p_score = self.align(ref_5p, _5p)
```

Comment:
after hunting down a bug i realized there's a drawback to this...

with this broken out 100% test coverage means every case is evaluated and tested. with this in a loop it means the loop has run once. i broke this out again so that we can clearly see which cases are covered. 

this was very helpful an hour ago when i made a subtle change to the 3' start position that made half of my tests redundant but coverage still showed 100% coverage. making this overly verbose again revealed exactly which test cases were overlapping and which behavior was untested. unfortunately it's not as beautiful as a nice little loop but the test coverage is more meaningful so i'm inclined to leave it verbose.